plugins/module_utils/network/sonic/config/lst/lst.py

```diff
@@ -0,0 +1,575 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The sonic_lst class
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from copy import deepcopy
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.cfg.base import (
+    ConfigBase,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    to_list,
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.facts.facts import Facts
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.utils import (
+    get_diff,
+    remove_empties,
+    update_states
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.sonic import (
+    edit_config,
+    to_request
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.formatted_diff_utils import (
+    __DELETE_CONFIG_IF_NO_SUBCONFIG,
+    get_new_config,
+    get_formatted_config_diff
+)
+
+LST_PATH = 'data/openconfig-lst-ext:lst'
+PATCH = 'patch'
+DELETE = 'delete'
+TEST_KEYS = [
+    {'lst_groups': {'name': ''}},
+    {'interfaces': {'id': ''}},
+    {'upstream_groups': {'group_name': ''}}
+]
+TEST_KEYS_generate_config = [
+    {'lst_groups': {'name': '', '__delete_op': __DELETE_CONFIG_IF_NO_SUBCONFIG}},
+    {'interfaces': {'id': '', '__delete_op': __DELETE_CONFIG_IF_NO_SUBCONFIG}},
+    {'upstream_groups': {'group_name': '', '__delete_op': __DELETE_CONFIG_IF_NO_SUBCONFIG}}
+]
+
+
+class Lst(ConfigBase):
+    """
+    The sonic_lst class
+    """
+
+    gather_subset = [
+        '!all',
+        '!min',
+    ]
+
+    gather_network_resources = [
+        'lst',
+    ]
+
+    def __init__(self, module):
+        super(Lst, self).__init__(module)
+
+    def get_lst_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        lst_facts = facts['ansible_network_resources'].get('lst')
+        if not lst_facts:
+            return {}
+        return lst_facts
+
+    def execute_module(self):
+        """ Execute the module
```

Comment:
Triple quotes for a docstring should have their own line. Descriptions following the triple quotes on one line occurs in multiple places throughout the file.
```suggestion
        """ 
        Execute the module
```